### PR TITLE
Swagger for diagnostic settings and association resource types

### PR DESCRIPTION
**Pull request by sankalpsoni**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by annatisch** [comment]
Thanks @sankalpsoni!
There's still a number of model validation errors:
https://travis-ci.org/Azure/azure-rest-api-specs/jobs/396031137

And linter errors - both missing examples and ARM violations:
https://travis-ci.org/Azure/azure-rest-api-specs/jobs/396031137

Are you able to take a look?

**Comment by sankalpsoni** [comment]
Are you talking errors which are reported in "Allowed Failures"? I am presuming that they should not block the sign off. 

What is the next step?

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json
Comment (ravbhatnagar) [inline specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json]: Shouldnt this be handled by the "default" error response? I dont think you need to add it explicitly

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json
Comment (ravbhatnagar) [inline specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json]: Since guestDiagnosticSettings is a top level tracked resource, supporting PATCH is required. Ideally it should support updating all patchable properties. But at the least, it should support updating tags.

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json
Comment (ravbhatnagar) [inline specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json]: Is this a long running operation?

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/examples/createOrUpdateGuestDiagnosticSettings.json
Comment (johnkemnetz) [inline specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/examples/createOrUpdateGuestDiagnosticSettings.json]: Can we capitalize the first letter of the OSes -- Windows/Linux?

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/examples/createOrUpdateGuestDiagnosticSettings.json
Comment (johnkemnetz) [inline specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/examples/createOrUpdateGuestDiagnosticSettings.json]: Don't you need the details for the kind? The workspace resource ID at least?

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/examples/createOrUpdateGuestDiagnosticSettingsAssociation.json
Comment (johnkemnetz) [inline specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/examples/createOrUpdateGuestDiagnosticSettingsAssociation.json]: Shouldn't our default example be an ARM VM?

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/examples/createOrUpdateGuestDiagnosticSettingsAssociation.json
Comment (johnkemnetz) [inline specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/examples/createOrUpdateGuestDiagnosticSettingsAssociation.json]: Maybe "centralITDefault" would be a better name to use as a sample.

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/examples/createOrUpdateGuestDiagnosticSettings.json
Comment (johnkemnetz) [inline specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/examples/createOrUpdateGuestDiagnosticSettings.json]: Suggestion: instead of SampleDiagSetting, use "productionMachineSetting" , which is more like a real customer scenario (one setting for all production machines)

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json
Comment (sankalpsoni) [reply specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json]: Done

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json
Comment (sankalpsoni) [reply specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json]: Done

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json
Comment (sankalpsoni) [reply specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json]: Nope. Should be completed in ms.

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/examples/createOrUpdateGuestDiagnosticSettings.json
Comment (sankalpsoni) [reply specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/examples/createOrUpdateGuestDiagnosticSettings.json]: Done

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/examples/createOrUpdateGuestDiagnosticSettings.json
Comment (sankalpsoni) [reply specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/examples/createOrUpdateGuestDiagnosticSettings.json]: Not for this private preview. Remember - multi-homing is not supported. Workspace resource id would be passed by extension

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/examples/createOrUpdateGuestDiagnosticSettingsAssociation.json
Comment (sankalpsoni) [reply specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/examples/createOrUpdateGuestDiagnosticSettingsAssociation.json]: Done

**Comment by johnkemnetz** [comment]
@sankalpsoni looks good to me.

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettingsAssociation_API.json
Comment (annatisch) [inline specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettingsAssociation_API.json]: This can be renamed to: `"guestDiagnosticsSettingsAssociation_List"` as the subscription ID is implicit.

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json
Comment (annatisch) [inline specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json]: This model name is incompatible with an already existing model name:
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/microsoft.insights/preview/2017-05-01-preview/diagnosticsSettings_API.json#L277

Do you intend to generate both specs into the same SDK? If so, this will need to be renamed (e.g. "GuestDiagnosticSettings"). Otherwise if this spec is intended to replace diagnosticSettings_API then it should be okay.

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json
Comment (annatisch) [inline specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json]: Could this please be renamed to just `"name: "parameters"`
We try to avoid API terms like "patch" in the clients :)

**Review by annatisch — CHANGES_REQUESTED** [review]

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json
Comment (annatisch) [inline specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json]: Please revert this to `"guestDiagnosticsSettings_ListByResourceGroup"`
This one didn't need to be changed.

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettingsAssociation_API.json
Comment (annatisch) [inline specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettingsAssociation_API.json]: Please revert this to "guestDiagnosticsSettingsAssociation_ListByResourceGroup"
This one didn't need to be changed.

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettingsAssociation_API.json
Comment (annatisch) [inline specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettingsAssociation_API.json]: Is there are reason you revert the name from "parameters" here? It's not an issue, but I'm curious.

**Review by annatisch — CHANGES_REQUESTED** [review]

**Comment by annatisch** [comment]
ping @sankalpsoni

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettingsAssociation_API.json
Comment (sankalpsoni) [reply specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettingsAssociation_API.json]: I do not remember. I have changed it back,

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettingsAssociation_API.json
Comment (sankalpsoni) [reply specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettingsAssociation_API.json]: Done

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json
Comment (sankalpsoni) [reply specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json]: Done

File: specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettingsAssociation_API.json
Comment (sankalpsoni) [reply specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettingsAssociation_API.json]: Now I know - it caused Model validation error. Done correctly now :)

**Comment by annatisch** [comment]
This PR is missing ARM sign-off. Since many days have passed, please let me know if you prefer to: 
1) follow up with ravbhatnagar and wait, or 
2) I can merge the PR without ARM’s team sign-off. Please note that even after merge, you should follow up with ARM team regarding ARM issues and fixes may impact the specs and/or generated SDKs. Thanks!

ping @ravbhatnagar

**Comment by ravbhatnagar** [comment]
Reviewed over PR and skype. Looks good.

**Review by annatisch — APPROVED** [review]

**Pull request merged**
